src/Cli.Tests/UpdateRuntimeTests.cs

```diff
@@ -0,0 +1,132 @@
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config);
+            Assert.IsNull(config!.Runtime!.GraphQL!.DepthLimit);
+
+            // Add Depth Limit
+            UpdateRuntimeOptions options = new(
+                depthLimit: maxDepthLimit,
+                config: _testConfigFile
+            );
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert after adding Depth Limit
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config!.Runtime!.GraphQL!.DepthLimit);
+            Assert.AreEqual(maxDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
+        }
+
+        /// <summary>
+        /// Test to update the current depth limit for GraphQL and removal the depth limit using -1.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(20, DisplayName = "Update current depth limit for GraphQL.")]
+        [DataRow(-1, DisplayName = "Remove depth limit from GraphQL.")]
+        public void TestUpdateDepthLimitForGraphQL(int? newDepthLimit)
+        {
+            int currentDepthLimit = 8;
+
+            // Generate Config with Depth Limit
+            RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config);
+            Assert.IsNotNull(config);
+            config = config with
+            {
+                Runtime = config!.Runtime! with
+                {
+                    GraphQL = config!.Runtime!.GraphQL! with
+                    {
+                        DepthLimit = currentDepthLimit
+                    }
+                }
+            };
+
+            _fileSystem!.AddFile(_testConfigFile, new MockFileData(config.ToJson()));
+
+            // Update Depth Limit
+            UpdateRuntimeOptions options = new(
+                depthLimit: newDepthLimit,
+                config: _testConfigFile
+            );
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert after adding Depth Limit
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config);
+
+            int? expectedDepthLimit = newDepthLimit == -1 ? null : newDepthLimit;
+            Assert.AreEqual(expectedDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
+
+            if (expectedDepthLimit is null)
+            {
+                Console.WriteLine(JObject.Parse(INITIAL_CONFIG));
+                Console.WriteLine(JObject.Parse(updatedConfig));
+                Assert.IsTrue(JToken.DeepEquals(JObject.Parse(INITIAL_CONFIG), JObject.Parse(updatedConfig)));
```

Comment:
a comment to add a little more detail to what you're validating here would be helpful. perhaps add `message:` param to `Assert.IsTrue` so the dev knows where to start investigating a fix if this test ever fails.